File: product-service/src/main/java/com/intexsoft/webshop/productservice/mapper/AttributeValueMapper.java
package com.intexsoft.webshop.productservice.mapper;

import com.intexsoft.webshop.productservice.dto.attributevalue.AttributeValueDto;
import com.intexsoft.webshop.productservice.model.AttributeValue;
import org.mapstruct.InjectionStrategy;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;

import java.util.List;
import java.util.Set;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING,
        injectionStrategy = InjectionStrategy.CONSTRUCTOR)
public interface AttributeValueMapper {
    List<AttributeValueDto> toAttributeDtoList(Set<AttributeValue> attributeValues);

    @Mapping(target = "id", source = "id")
    @Mapping(target = "value", source = "value")
    @Mapping(target = "attributeId", source = "attributeValue.attribute.id")
    AttributeValueDto attributeValueToAttributeValueDto(AttributeValue attributeValue);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/mapper/AttributeValueMapper.kt
package com.intexsoft.webshop.productservicekt.mapper

import com.intexsoft.webshop.productservicekt.dto.attributevalue.AttributeValueDto
import com.intexsoft.webshop.productservicekt.model.AttributeValue
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    injectionStrategy = InjectionStrategy.CONSTRUCTOR
)
interface AttributeValueMapper {
    @Mapping(target = "id", source = "id")
    @Mapping(target = "value", source = "value")
    @Mapping(target = "attributeId", source = "attributeValue.attribute.id")
    fun attributeValueToAttributeValueDto(attributeValue: AttributeValue): AttributeValueDto
}